src/Traits/GenericActionsTrait.php

```diff
@@ -57,23 +57,119 @@ protected function loadFilteredChildren(string $id): array
         return $this->paginate($children->order([], true), ['order' => ['Trees.tree_left']])->toList();
     }
 
+    /**
+     * Dispatch before object load hook events.
+     *
+     * @param string $uname Object uname.
+     * @param string $type Object type.
+     * @return \Cake\Http\Response|null The resulting response of the events.
+     */
+    protected function dispatchBeforeLoadEvent(string $uname, string $type): ?Response
+    {
+        $event = $this->dispatchEvent(sprintf('Controller.beforeObjectLoad:%s', $uname), compact('uname', 'type'));
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
+
+        $event = $this->dispatchEvent(sprintf('Controller.beforeObjectLoad:%s', $type), compact('uname', 'type'));
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
+
+        $event = $this->dispatchEvent('Controller.beforeObjectLoad', compact('uname', 'type'));
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
+
+        return null;
+    }
+
+    /**
+     * Dispatch after object load hook events.
+     *
+     * @param string $uname Object uname.
+     * @param string $type Object type.
+     * @param array $data Objects data to pass to callback.
+     * @return \Cake\Http\Response|null The resulting response of the events.
+     */
+    protected function dispatchAfterLoadEvent(string $uname, string $type, array $data): ?Response
+    {
+        $event = $this->dispatchEvent(sprintf('Controller.afterObjectLoad:%s', $uname), $data);
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
+
+        $event = $this->dispatchEvent(sprintf('Controller.afterObjectLoad:%s', $type), $data);
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
+
+        $event = $this->dispatchEvent('Controller.afterObjectLoad', $data);
+        if ($event->getResult() !== null) {
+            return $event->getResult();
+        }
```

Comment:
Same here. I'd just dispatch the `Controller.afterObjectLoad` event, and then leave the listener decide whether or not the `uname` and `type` are relevant for them, and if not return without doing anything.